packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,422 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import StoreKit
+
+#if os(iOS)
+  import Flutter
+#elseif os(macOS)
+  import FlutterMacOS
+#endif
+
+public class InAppPurchasePlugin: NSObject, FlutterPlugin, InAppPurchaseAPI {
+  private let receiptManager: FIAPReceiptManager
+  private var productsCache: NSMutableDictionary = [:]
+  private var paymentQueueDelegateCallbackChannel: FlutterMethodChannel?
+  // note - the type should be FIAPPaymentQueueDelegate, but this is only available >= iOS 13,
+  // FIAPPaymentQueueDelegateonly gets set/used in registerPaymentQueueDelegateWithError or removePaymentQueueDelegateWithError, which both are ios13+ only
+  private var paymentQueueDelegate: Any? = nil
+  private var requestHandlers = Set<FIAPRequestHandler>()
+  private var handlerFactory: ((SKRequest) -> FIAPRequestHandler)
+  // TODO(louisehsu): Once tests are migrated to swift, we can use @testable import, and make theses vars private again
+  @objc
+  public var registrar: FlutterPluginRegistrar?
+  @objc
+  public var paymentQueueHandler: FIAPaymentQueueHandler?
+  @objc
+  public var transactionObserverCallbackChannel: FlutterMethodChannel?
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    #if os(iOS)
+      let messenger = registrar.messenger()
+    #endif
+    #if os(macOS)
+      let messenger = registrar.messenger
+    #endif
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/in_app_purchase",
+      binaryMessenger: messenger)
+    let instance = InAppPurchasePlugin(registrar: registrar)
+    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addApplicationDelegate(instance)
+    SetUpInAppPurchaseAPI(messenger, instance)
+  }
+
+  @objc
+  public init(receiptManager: FIAPReceiptManager) {
+    self.receiptManager = receiptManager
+    self.requestHandlers = Set<FIAPRequestHandler>()
+    self.handlerFactory = {
+      FIAPRequestHandler(request: $0)
+    }
+    super.init()
+  }
+
+  @objc
+  // TODO(louisehsu): this initializer is for tests only. Make this private after the tests are moved to swift.
+  public convenience init(
```

Comment:
Interesting! Okay, i changed the designated initizalizers, but I cant make it into one init, since when it gets bridged to objc it forces the default value to be a normal param.

Reply:
You may also want to take `paymentQueueHandler` as another param for the designated init. We want other `init`s to call this designated `init`, so that `paymentQueueHandler` can be a `let` rather than a `var`. 

```
let p1: Int
let p2: Int
let p3: Int?
init(p1: Int, p2: Int, p3: Int?) {}

convenience init(p1: Int, p2: Int) {
  self.init(p1: p1, p2: p2, p3: nil)
}

convenience init(p1: Int, p3: Int?) {
  self.init(p1: p1, p2: 123, p3: p3)
}
```

This kind of setup avoids haven't to use `var` in fields that don't actually change.

Reply:
discussed offline - paymentQueueHandler cannot be a let var as it requires self to be initialized